sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/ResourceEvaluationStage.java

```diff
@@ -0,0 +1,182 @@
+package com.mesosphere.sdk.offer.evaluate;
+
+import com.google.protobuf.TextFormat;
+import com.mesosphere.sdk.offer.*;
+import org.apache.mesos.Protos;
+import org.apache.mesos.Protos.Resource;
+import org.apache.mesos.Protos.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * This class evaluates an offer against a given {@link OfferRequirement}, ensuring that it contains a sufficient amount
+ * or value of the supplied {@link Resource}, and creating a {@link ReserveOfferRecommendation} or
+ * {@link UnreserveOfferRecommendation} where necessary.
+ */
+public class ResourceEvaluationStage implements OfferEvaluationStage {
+    private static final Logger logger = LoggerFactory.getLogger(ResourceEvaluationStage.class);
+
+    private ResourceRequirement resourceRequirement;
+    // Resources frequently have necessary task affinities -- a dynamic port, for example, will need to modify the
+    // TaskInfo it belongs to. We express that affinity with this field, which if left null means that this evaluation
+    // stage modifies the ExecutorInfo instead.
+    private String taskName;
+
+    public ResourceEvaluationStage(Resource resource, String taskName) {
+        this.resourceRequirement = new ResourceRequirement(resource);
+        this.taskName = taskName;
+    }
+
+    public ResourceEvaluationStage(Resource resource) {
+        this(resource, null);
+    }
+
+    protected void setResourceRequirement(ResourceRequirement resourceRequirement) {
+        this.resourceRequirement = resourceRequirement;
+    }
+
+    protected ResourceRequirement getResourceRequirement() {
+        return resourceRequirement;
+    }
+
+    protected Optional<String> getTaskName() {
+        return Optional.ofNullable(taskName);
+    }
+
+    @Override
+    public void evaluate(
+            MesosResourcePool offerResourcePool,
+            OfferRequirement offerRequirement,
+            OfferRecommendationSlate offerRecommendationSlate) throws OfferEvaluationException {
+        ResourceRequirement resourceRequirement = getResourceRequirement();
+        Optional<MesosResource> mesosResourceOptional = offerResourcePool.consume(resourceRequirement);
+        if (!mesosResourceOptional.isPresent()) {
+            throw new OfferEvaluationException(String.format(
+                    "Failed to satisfy resource requirement: %s",
+                    TextFormat.shortDebugString(resourceRequirement.getResource())));
+        }
+
+        final MesosResource mesosResource = mesosResourceOptional.get();
+        Resource fulfilledResource = getFulfilledResource(mesosResource);
+        if (resourceRequirement.expectsResource()) {
+            logger.info("Expects Resource");
```

Comment:
Maybe port the log messages from the current/old OfferEvaluator? They made it easier to tell what was/wasn't passing evaluation. See e.g. old OfferEvaluator.java:L291 onwards.